src/main/java/org/apache/flink/benchmark/MultiInputCheckpointingTimeBenchmark.java

```diff
@@ -0,0 +1,189 @@
+                            .slotSharingGroup("source-small-records")
+                            .rebalance();
+
+            DataStream<Record> source2 =
+                    env.fromSource(
+                                    new RecordSource(
+                                            Integer.MAX_VALUE, (int) BIG_RECORD_SIZE.getBytes()),
+                                    noWatermarks(),
+                                    RecordSource.class.getName())
+                            .slotSharingGroup("source-big-records")
+                            .rebalance();
+
+            source1.connect(source2)
+                    .map(
+                            new CoMapFunction<Record, Record, Record>() {
+                                @Override
+                                public Record map1(Record record) throws Exception {
+                                    return record;
+                                }
+
+                                @Override
+                                public Record map2(Record record) throws Exception {
+                                    return record;
+                                }
+                            })
+                    .name("co-map")
+                    .slotSharingGroup("map-and-sink")
+                    .addSink(new SlowDiscardSink<>())
+                    .slotSharingGroup("map-and-sink");
+
+            final StreamGraph streamGraph = env.getStreamGraph(false);
+            final JobGraph jobGraph = streamGraph.getJobGraph();
+            final List<JobVertex> vertices = jobGraph.getVerticesSortedTopologicallyFromSources();
+
+            final JobClient jobClient = env.executeAsync();
+            jobID = jobClient.getJobID();
+            CommonTestUtils.waitForAllTaskRunning(miniCluster, jobID, false);
+            List<JobVertexID> sourceIds =
+                    Arrays.asList(vertices.get(0).getID(), vertices.get(1).getID());
+            BackpressureUtils.waitForBackpressure(
+                    jobID, sourceIds, miniCluster.getRestAddress().get());
+            Thread.sleep(DEBLOATING_STABILIZATION_PERIOD);
+        }
+
+        @Override
+        protected Configuration createConfiguration() {
+            final Configuration config = super.createConfiguration();
+            config.set(ExecutionCheckpointingOptions.ENABLE_UNALIGNED, false);
+            config.set(TaskManagerOptions.MEMORY_SEGMENT_SIZE, START_MEMORY_SEGMENT_SIZE);
+            config.set(TaskManagerOptions.MIN_MEMORY_SEGMENT_SIZE, MIN_MEMORY_SEGMENT_SIZE);
+            config.set(TaskManagerOptions.BUFFER_DEBLOAT_ENABLED, true);
+            config.set(TaskManagerOptions.BUFFER_DEBLOAT_TARGET, DEBLOATING_TARGET);
+            config.set(
+                    TaskManagerOptions.BUFFER_DEBLOAT_PERIOD, Duration.of(200, ChronoUnit.MILLIS));
+            config.set(TaskManagerOptions.BUFFER_DEBLOAT_SAMPLES, 5);
+            return config;
```

Comment:
Maybe even this? Isn't it the same as `ALIGNED` mode from the singleInput benchmark?